include/pmacc/math/operation/And.hpp

```diff
@@ -0,0 +1,43 @@
+/* Copyright 2023 Brian Marre
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc/types.hpp"
+
+#include <cstdint>
+
+namespace pmacc::math::operation
+{
+    struct And
+    {
+        HDINLINE void operator()(uint32_t& destination, uint32_t const& source) const
```

Comment:
more a question to @psychocoderHPC:
Years ago, I learned that passing by reference is only useful for larger data. I would consider `uint32_t`to be rather small and would always have chosen passing a copy. What is your opinion on that?

Reply:
I followed the examples of the general add/sub/mul, they use a template type isntead of hard coded bit width maybe explaining the difference in interface

Reply:
@PrometheusPi is right, the second parameter should not be a reference. But if other operators are written equally we should keep it like that and change it independently.

Reply:
Okay, the let's do this later*

*=never ;)